[CS2113T-T12-4] RestaurantBuddy

jerrelllzw opened the pull request:
RestaurantBuddy helps restaurant managers keep track of their employees and inventory. It is optimized for CLI users so that frequent tasks can be done faster by typing in commands.

jerrelllzw committed:
Added details to the docs/AboutUs.md page

kairoskoh committed:
Add details to AboutUs.md

leecheokfeng committed:
Add name

Cesare4869 committed:
changed name

Cesare4869 committed:
Merge pull request #3 from leecheokfeng/leeCheokFeng-AboutUs

Add individual name and github username

kairoskoh committed:
Resolve conflicts on AboutUs.md

jerrelllzw committed:
Merge pull request #1 from jerrelllzw/jerrellLow-AboutUs

Added details to the docs/AboutUs.md page

Cesare4869 committed:
resolve conflicts

jerrelllzw committed:
Merge pull request #2 from kairoskoh/kairosKoh-AboutUs

Add details to AboutUs.md

Cesare4869 committed:
merged conflict zehang

Cesare4869 committed:
Merge pull request #4 from Cesare4869/Zehang-AboutUs

Merged

Cesare4869 committed:
new change in aboutus

Cesare4869 committed:
Merge pull request #15 from Cesare4869/Zehang-AboutUs

new change in aboutus

Cesare4869 committed:
copy some ip code to tp

Cesare4869 committed:
include parser and disable storage, now can only do exit and false command

Cesare4869 committed:
Merge pull request #16 from Cesare4869/Zehang-AboutUs

Add parser

kairoskoh committed:
Add Employee.java
Add EmployeeList.java
Add EmployeeParser.java

kairoskoh committed:
Add basic feature for Employee.java

kairoskoh committed:
Add basic feature for EmployeeList.java

kairoskoh committed:
Add basic feature for EmployeeParser.java

kairoskoh committed:
Add EmployeeParserTest.java

kairoskoh committed:
Merge pull request #17 from kairoskoh/branch-week8-kairos

Branch week8 kairos

kairoskoh committed:
Add new packages for ingredients and menu

kairoskoh committed:
Merge pull request #18 from kairoskoh/branch-week8-meeting

Add new packages for ingredients and menu

kairoskoh committed:
Add remove and list functions for employee

jerrelllzw committed:
Added MenuItem class and its related functions (add, remove, list)

Cesare4869 committed:
add storage in main method

Cesare4869 committed:
add storage for all classed in main

leecheokfeng committed:
Create ingredient class and functions to list/remove/add

kairoskoh committed:
Merge pull request #19 from kairoskoh/branch-week8-kairos-employee

Add remove and list functions for employee

kairoskoh committed:
Merge pull request #23 from leecheokfeng/branch-week8-ingredient

Create ingredient class and functions to list/remove/add

kairoskoh committed:
Merge pull request #22 from Cesare4869/master

Add storage for employee, menu and ingredient

jerrelllzw committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T12-4/tp into Menu

# Conflicts:
#	src/main/java/seedu/duke/Duke.java

kairoskoh committed:
Merge pull request #20 from jerrelllzw/Menu

Added MenuItem class and its related functions (add, remove, list)

kairoskoh committed:
Refactor code to reduce number of lines of code in main method

kairoskoh committed:
Merge pull request #25 from kairoskoh/branch-week9-meeting

Refactor code to reduce number of lines of code in main method

leecheokfeng committed:
Refactor ingredientParser to use printLine

leecheokfeng committed:
Merge pull request #26 from leecheokfeng/branch-week8-ingredient

Refactor ingredientParser to use printLine

kairoskoh committed:
Add ascii art for welcome message

kairoskoh committed:
Remove additional print lines to reduce clutter on command line

kairoskoh committed:
Add fixes for whitespaces inputs

Cesare4869 committed:
fix storage problem

jerrelllzw committed:
Merge branch 'Menu'

jerrelllzw committed:
- added jUnit test for MenuParser class
- added code in gradle to enable assertions

jerrelllzw committed:
added assertions into deleteMenu method in MenuParser class

jerrelllzw committed:
added comments to the assertions

jerrelllzw committed:
-added MenuUi class for menu related messages to be printed
-added invalid index error handling

Fixes #29

jerrelllzw committed:
-added error handling for command inputs with invalid parameters
-fully fixed the delete-menu bug where target index is invalid

Fixes #31

jerrelllzw committed:
reformatted code

jerrelllzw committed:
-reformatted code

kairoskoh committed:
Add logging for employeeparser class

Cesare4869 committed:
add assertion in storage

kairoskoh committed:
Add logging and assert for employeeparser class

leecheokfeng committed:
Add exception handling to ingredientParser

leecheokfeng committed:
Extract UI into ingredientUI class

jerrelllzw committed:
Merge pull request #37 from leecheokfeng/branch-week9-ingredient-exceptions

Branch week9 ingredient exceptions

jerrelllzw committed:
Merge pull request #34 from kairoskoh/branch-week9-kairos

Branch week9 kairos

jerrelllzw committed:
Update MainParser.java

jerrelllzw committed:
Merge pull request #33 from Cesare4869/master

Fix storage and add assertions

jerrelllzw committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T12-4/tp into Errors

# Conflicts:
#	src/main/java/seedu/duke/storage/Storage.java

jerrelllzw committed:
Merge pull request #32 from jerrelllzw/Errors

Added error handling and menu UI

jerrelllzw committed:
-corrected checkstyle errors

jerrelllzw committed:
Merge pull request #38 from jerrelllzw/Errors

-corrected checkstyle errors

jerrelllzw committed:
fixed a bug in removeMenu method where index is always invalid

jerrelllzw committed:
Merge pull request #43 from jerrelllzw/Errors

fixed a bug in removeMenu method where index is always invalid

jerrelllzw committed:
-fixed menu related bugs
-improved menu ui messages

jerrelllzw committed:
-add edit menu price function

jerrelllzw committed:
changed price attribute from string type to double type

jerrelllzw committed:
-started on user guide
-renamed menu related classes for easier understanding

jerrelllzw committed:
-renamed menu class in test

jerrelllzw committed:
fixed previous commit error

jerrelllzw committed:
fixed checkstyle error

jerrelllzw committed:
added promotion function

jerrelllzw committed:
added more test methods for dish parser

jerrelllzw committed:
added author comments

jerrelllzw committed:
added draft version of user guide

jerrelllzw committed:
Merge pull request #44 from jerrelllzw/Promotion

Promotion

jerrelllzw committed:
changed seperator

jerrelllzw committed:
Merge pull request #45 from jerrelllzw/Promotion

changed seperator

Cesare4869 committed:
finance and DG

Cesare4869 committed:
docs ug, dg, finance, dish merged

Cesare4869 committed:
Merge pull request #46 from Cesare4869/master

Have UG/DG, Finance and Dish

kairoskoh committed:
Add changes to developguide.md on employee package

kairoskoh committed:
Merge pull request #47 from kairoskoh/master

Add changes to developguide.md on employee package

jerrelllzw committed:
Merge branch 'Promotion'

jerrelllzw committed:
edited developer guide

jerrelllzw committed:
Merge pull request #48 from jerrelllzw/master

edited developer guide

kairoskoh committed:
Add introduction and table of contents

kairoskoh committed:
Merge pull request #50 from kairoskoh/master

Add introduction and table of contents

kairoskoh committed:
Add employee UI

kairoskoh committed:
Merge pull request #52 from kairoskoh/branch-week10-kairos

Add employee UI

jerrelllzw committed:
-added hyper links to content page for user guide
-added finance methods to user guide

jerrelllzw committed:
updated user guide

jerrelllzw committed:
Merge pull request #53 from jerrelllzw/Docs

Docs

jerrelllzw committed:
Update UserGuide.md

jerrelllzw committed:
added indexing

jerrelllzw committed:
added more sections and changed formatting

jerrelllzw committed:
Merge branch 'master' into Docs

jerrelllzw committed:
Merge pull request #58 from jerrelllzw/Docs

Docs

leecheokfeng committed:
Add price attribute and check expiry function to ingredient

leecheokfeng committed:
Add feature implementation for finding expiry

leecheokfeng committed:
Merge pull request #62 from leecheokfeng/branch-week10-ingredient

Add ingredient price, find expired ingredients

leecheokfeng committed:
Update UG to include ingredient price and expiry feature

leecheokfeng committed:
Merge pull request #63 from leecheokfeng/branch-week11-ingredient

Update UG to include ingredient price and expiry feature

Cesare4869 committed:
change dg

Cesare4869 committed:
Merge pull request #64 from Cesare4869/master

change dg

jerrelllzw committed:
edited dg (user stories, target)

jerrelllzw committed:
Merge pull request #65 from jerrelllzw/master

edited dg (user stories, target)

kairoskoh committed:
Add purpose and how to use section under user guide

kairoskoh committed:
Add employee status and salary

jerrelllzw committed:
edited dg (added non funtioncal requirements and glossary)

jerrelllzw committed:
Merge pull request #67 from jerrelllzw/Docs

edited dg (added non funtioncal requirements and glossary)

jerrelllzw committed:
- removed error message on startup
- added lines to wrap storage error messages

jerrelllzw committed:
Merge pull request #68 from jerrelllzw/Docs

- removed error message on startup

jerrelllzw committed:
fixed dish storage bug

jerrelllzw committed:
Merge pull request #69 from jerrelllzw/Docs

fixed dish storage bug

kairoskoh committed:
Remove contents of Duke.txt for storage

kairoskoh committed:
Update UI class

kairoskoh committed:
Add exceptions when handling invalid input

jerrelllzw committed:
fixed bug

jerrelllzw committed:
Merge pull request #70 from jerrelllzw/Docs

fixed bug

kairoskoh committed:
Add more logs

kairoskoh committed:
Merge pull request #71 from kairoskoh/branch-week11-kairos

Branch week11 kairos

WaiKit-nus commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,211 @@
+The storage component has a `Storage` class which can load data from the file and save data into the file with the 
+methods to encode and decode data.  
+
+## Implementation  
```
For the implementation part, I feel that a step by step guide should be done. Common features like the Remove Dish Feature and Remove Ingredient Feature can be explained as 1 entity.

WaiKit-nus replied:
But other than that, I like how all the features are summarized:)

WaiKit-nus replied:
You can also include some sequence diagram/class diagram to further help you explain the feature:)
Example:
![image](https://user-images.githubusercontent.com/80661630/139193781-159db55b-0312-4a02-b43c-41a22ee35ba7.png)

WaiKit-nus commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,211 @@
+  * [Ingredient Component](#ingredient-component)
+  * [Dish Component](#dish-component)
+  * [Storage Component](#storage-component)
+* [Implementation](#implementation)
```
There are subcomponents inside the implementation parts. Maybe its better to add this in like how you did for the Design Components.
![image](https://user-images.githubusercontent.com/80661630/139192500-1cef48c4-4774-415d-81cb-68fa4c723ec7.png)
![image](https://user-images.githubusercontent.com/80661630/139192566-093fc841-d075-4b57-b52a-5123567ae226.png)

WaiKit-nus commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,211 @@
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Design
 
+### Architecture
```
I think adding a high level design diagram for the Architecture component will make your DG look nicer and more intuitive. Can do something like this:
![image](https://user-images.githubusercontent.com/80661630/139192817-149b9625-16e9-4552-b281-ab43e4826f31.png)

WaiKit-nus commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,211 @@
+* `Ingredient` stores the name and quantity of a particular ingredient used by the restaurant, as well as methods to 
+retrieve ingredient data.  
+
+* `IngredientList` is an array list of Ingredient.  
```
I think its abit confusing on what IngredientList should be. Maybe you can paraphrase to be like "IngredientList is a class that contains an array list of ingredients?" Not sure if you mean it that way, just giving you a suggestion:) 
![image](https://user-images.githubusercontent.com/80661630/139193151-c3e3f00e-16e1-42fa-9914-c2ae0643d6f9.png)

WaiKit-nus commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,211 @@
+* `Dish` stores the name, price and discount of a certain dish sold by the restaurant and the methods used to retrieve 
+them.
+
+* `Menu` is an arrayList of Dish.
```
Same as the IngredientList, I think you can explain it better :)

WaiKit-nus commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,211 @@
+program has, it gets the item from them. Then, it encodes the string representation of the item, and writes it into the 
+file.
+
+The three lists are stored in sequence as shown below,
```
Just a note here, it's not completed here. Don't forget about this part:)

WaiKit-nus reviewed (commented):
Overview:
Format of the DG looks good! Clickable links are all okay, keep up the good work!

vincentlauhl commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,211 @@
+  * [Instructions for Manual Testing](#instructions-for-manual-testing)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Design
```
Perhaps you can add an architecture diagram to give a higher level overview of how each component interacts with each other.

vincentlauhl commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,211 @@
+
+### Find Expired Ingredients Feature
+The mechanism of finding expired ingredients is facilitated by `IngredientParser` as well as the Java `LocalDate` 
+class. `IngredientParser` converts the input String into a `LocalDate` object and calls a method from `IngredientUI`.
+This method loops through the current `IngredientList` and finds all instances of `Ingredient` with an expiry date
+before the input `LocalDate` object. It then displays to the user the list of ingredients that are expired, or informs
```
Should the UI be responsible for finding the expired ingredients?

vincentlauhl commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,211 @@
+list which is generated.
+
+The three types of new item are identified before decoding it, and the decoding methods are worked on at the 
+same time.
```
Perhaps the part "the decoding methods are worked on at the same time" can be further elaborated with a diagram.

vincentlauhl commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,211 @@
+instance of `Ingredient`, and adds it to the existing instance of `IngredientList`. It then calls a method from 
+`IngredientUI` to display a confirmation message to the user.
+
+### List Ingredients Feature
```
Perhaps you can add class and sequence diagrams to each features

Cesare4869 committed:
31/10

Cesare4869 committed:
Merge https://github.com/AY2122S1-CS2113T-T12-4/tp

Cesare4869 committed:
add finance part in DG and edit the storage part

Cesare4869 committed:
Merge pull request #118 from Cesare4869/master

add fiannce part in DG and edit the storage part

Cesare4869 committed:
add diagrams folder

Cesare4869 committed:
Merge pull request #119 from Cesare4869/master

add diagrams folder

Cesare4869 committed:
edit ug-addfinance and change the finance date type to LocalDate also fix storage

Cesare4869 committed:
Merge pull request #120 from Cesare4869/master

edit ug-addfinance and change the finance date type to LocalDate also…

jerrelllzw committed:
renamed list-menu to list-dish

jerrelllzw committed:
fixed bugs according to ped comments

jerrelllzw committed:
Merge pull request #121 from jerrelllzw/Bugs

fixed bugs according to ped comments for dish classes

jerrelllzw committed:
-added decimal formatting for dish prices
-edited discount-dish messages
-updated user guide accordingly

jerrelllzw committed:
Merge pull request #122 from jerrelllzw/Bugs

edited code based on ped comments

jerrelllzw committed:
added javadoc comments for dish related classes

jerrelllzw committed:
Merge pull request #123 from jerrelllzw/Bugs

added javadoc comments for dish related classes

jerrelllzw committed:
-added team picture folder
-edited about us doc
-added contents indexing for developer guide
-added explanations for dish related classes in developer guide

jerrelllzw committed:
Merge pull request #124 from jerrelllzw/Bugs

edited developer guide

jerrelllzw committed:
-added info into team docs
-added diagrams for dish component and add dish

jerrelllzw committed:
Merge pull request #125 from jerrelllzw/Bugs

-added info into team docs

kairoskoh committed:
Removed data from Duke.txt

kairoskoh committed:
Add authorship comments

jerrelllzw committed:
added dish diagrams

kairoskoh committed:
Added PPP

kairoskoh committed:
Merge pull request #127 from kairoskoh/master

Add PPP

kairoskoh committed:
Add javadoc

kairoskoh committed:
Merge pull request #128 from kairoskoh/master

Add javadoc

kairoskoh committed:
Fixes #78

kairoskoh committed:
Merge pull request #129 from kairoskoh/master

Fixes #78

kairoskoh committed:
Fixes #79

kairoskoh committed:
Merge pull request #130 from kairoskoh/master

Fixes #79

kairoskoh committed:
Fixes #82

kairoskoh committed:
Merge pull request #131 from kairoskoh/master

Fixes #82

kairoskoh committed:
Add minor changes to user guide

kairoskoh committed:
Merge pull request #132 from kairoskoh/master

Add minor changes to user guide

leecheokfeng committed:
Added JUnit Tests

leecheokfeng committed:
Update UG description based on PE-D
Fixes #117

leecheokfeng committed:
Add UI to describe proper command syntax to users
Fixes #109

Cesare4869 committed:
add PPP and change code in finance and DG

Cesare4869 committed:
Merge pull request #133 from Cesare4869/master

add PPP and change code in finance and DG

jerrelllzw committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T12-4/tp into Bugs

jerrelllzw committed:
-PPP for jerrell

jerrelllzw committed:
Merge pull request #134 from jerrelllzw/Bugs

PPP - jerrell

jerrelllzw committed:
Update UserGuide.md

leecheokfeng committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T12-4/tp into branch-week12-ingredient

leecheokfeng committed:
Add PPP for Cheok Feng and edit AboutUs

leecheokfeng committed:
Handle ingredient errors for negative or non integer values

Cesare4869 committed:
change portfilo and add test and javadoc for finance

Cesare4869 committed:
Merge branch 'master' into master

Cesare4869 committed:
Merge pull request #135 from Cesare4869/master

change portfilo and add test and javadoc for finance

leecheokfeng committed:
Make parser remove whitespaces from input data
Fixes #100

leecheokfeng committed:
Handle case when user enters blank ingredient name

leecheokfeng committed:
Update UG examples for ingredient

leecheokfeng committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T12-4/tp into branch-week12-ingredient

# Conflicts:
#	docs/AboutUs.md

leecheokfeng committed:
Merge pull request #136 from leecheokfeng/branch-week12-ingredient

Branch ingredient exceptions, UG, PPP

kairoskoh committed:
Add changes to quick start in user guide

kairoskoh committed:
Add details in about us

kairoskoh committed:
Add changes to user guide

kairoskoh committed:
Add help command

kairoskoh committed:
Add help to user guide

jerrelllzw committed:
edited documentation

jerrelllzw committed:
Merge pull request #137 from jerrelllzw/final

edited documentation

kairoskoh committed:
Add minor changes to checkstyle

kairoskoh committed:
Merge pull request #138 from kairoskoh/branch-week12-kairos

Add help command and checkstyle fixes

kairoskoh committed:
Add changes to i/o tests

kairoskoh committed:
Add input/output for help command

kairoskoh committed:
Add i/o tests to gitignore

kairoskoh committed:
Remove text-ui-test

kairoskoh committed:
Revert "Remove text-ui-test"

This reverts commit d8bd525bcf3e172458c7d5e30b7fc7a3f438027e.

kairoskoh committed:
Remove contents of text-ui-test

kairoskoh committed:
Merge pull request #143 from kairoskoh/branch-week12-kairos

Remove contents in text-ui-test

leecheokfeng committed:
Add Javadoc for ingredient

leecheokfeng committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T12-4/tp into branch-week12-ingredient

leecheokfeng committed:
Merge pull request #144 from leecheokfeng/branch-week12-ingredient

Add Javadoc for ingredient

kairoskoh committed:
Add employee class diagram

leecheokfeng committed:
Add diagrams to DG for ingredient

leecheokfeng committed:
Merge pull request #145 from leecheokfeng/branch-week12-ingredient

Add diagrams to DG for ingredient

kairoskoh committed:
Merge pull request #146 from kairoskoh/branch-week12-dg

Add employee class diagram

kairoskoh committed:
Add changes to README.md

jerrelllzw committed:
Update jerrelllzw.md

kairoskoh committed:
Merge pull request #147 from kairoskoh/branch-week12-dg

Add changes to README.md

leecheokfeng committed:
Update ingredient commands in command summary and help

leecheokfeng committed:
Merge pull request #148 from leecheokfeng/branch-week12-ingredient

Update ingredient commands in command summary and help

jerrelllzw committed:
fixed lifeline for dish sequence diagram

jerrelllzw committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-T12-4/tp into final

jerrelllzw committed:
Merge pull request #149 from jerrelllzw/final

Final

Cesare4869 committed:
Change the diagrams for finance

Cesare4869 committed:
Merge pull request #150 from Cesare4869/master

Change the diagrams for finance

Cesare4869 committed:
correct small issue in show-finance

Cesare4869 committed:
Merge pull request #151 from Cesare4869/master

correct small issue in show-finance

kairoskoh committed:
Add sequence diagram to developer guide

kairoskoh committed:
Merge pull request #152 from kairoskoh/branch-week12-dg

Add sequence diagram to developer guide

kairoskoh committed:
Fix checkstyle errors

kairoskoh committed:
Merge pull request #153 from kairoskoh/branch-week12-dg

Fix checkstyle errors
